broker-daemon/models/block-order.js

```diff
@@ -476,6 +476,100 @@ class BlockOrder {
 
     return BlockOrder.fromStorage(blockOrderId, value)
   }
+
+  /**
+   * Grabs all orders for a range of block orders
+   * @param {sublevel} store
+   * @param {string} first
+   * @param {string} last
+   * @returns {void}
+   */
+  static async getOrdersForRange (store, first, last) {
+    return getRecords(
+      store,
+      (key, value) => {
+        const { order, state, error, dates } = JSON.parse(value)
+        return { order: Order.fromObject(key, order), state, error, dates }
+      },
+      // limit the orders we retrieve to those that belong to this blockOrder, i.e. those that are in
+      // its prefix range.
+      Order.rangeForIds(first, last)
+    )
+  }
+
+  static async getFillsForRange (store, first, last) {
+    return getRecords(
+      store,
+      (key, value) => {
+        const { fill, state, error, dates } = JSON.parse(value)
+        return { fill: Fill.fromObject(key, fill), state, error, dates }
+      },
+      // limit the fills we retrieve to those that belong to this blockOrder, i.e. those that are in
+      // its prefix range.
+      Fill.rangeForIds(first, last)
+    )
+  }
+
+  /**
+   * @param {Array<Order>} orders
+   * @returns {Object} res
+   * @returns {Big} res.inbound
+   * @returns {Big} res.outbound
+   */
+  static async activeAmountsForOrders (orders) {
```

Comment:
since this is so duplicative of `BlockOrder#activeOutboundAmount`, is there a way to refactor this or that method to use the same code paths?

Reply:
Although it seems duplicative, its for different models. I felt like making this more abstract would make it harder to read

Reply:
I tried doing something like

```
activeAmountsForModel (data, allowedStates)
```

and then trying to make the logic for `if EXECUTING` look nicer. I wasn't happy with it

Reply:
Moving these to the Fill/Order, respectively, might be a better solution just cause they are probably weird to have on the BlockOrder itself

Reply:
I was thinking that `activeInboundAmount` and `activeOutboundAmount` would depend on these methods.

Reply:
That works for me